syntax = "proto3";
import "common.proto";
package dasproto;

message HandleList {
    repeated string list = 1;
    int32 request_type = 2;
    int64 hebbian_network = 3;
    string context = 4;
}

message HandleCount {
    map<string, uint32> map = 1;
    int32 request_type = 2;
    int64 hebbian_network = 3;
    string context = 4;
}

message ImportanceList {
    repeated float list = 1;
}

service AttentionBroker {
    rpc ping (Empty) returns (Ack) {}
    rpc stimulate (HandleCount) returns (Ack) {}
    rpc correlate (HandleList) returns (Ack) {}
    rpc get_importance (HandleList) returns (ImportanceList) {}
}
